web/packages/teleport/src/Navigation/RecentHistory.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { matchPath } from 'react-router';
+import { NavLink } from 'react-router-dom';
+
+import { Flex, Text, Box, H4 } from 'design';
+import { Cross } from 'design/Icon';
+
+import { TeleportFeature } from 'teleport/types';
+import { useFeatures } from 'teleport/FeaturesContext';
+
+import { getSubsectionStyles } from './SideNavigation/Section';
+import { SidenavCategory } from './SideNavigation/categories';
+
+/**
+ * RecentHistoryItem is an item in the recent history list, this is the data format stored in the recent history kept in localStorage.
+ */
+export type RecentHistoryItem = {
+  category?: SidenavCategory;
+  title: string;
+  route: string;
+  exact?: boolean;
+};
+
+function getIconForRoute(
+  features: TeleportFeature[],
+  route: string
+): (props) => JSX.Element {
+  const feature = features.find(feature =>
+    matchPath(route, {
+      path: feature?.route?.path,
+      exact: false,
+    })
+  );
+
+  return feature?.navigationItem?.icon || feature?.topMenuItem?.icon;
```

Comment:
this means it _could_ return undefined if neither of these values exist (which would be the fault of dev of course) but later when we use this to render the item it seems to think we that we will always have an `Icon`. Can we add a conditional render for this? or a default icon or something?